db/migrate/20221216171738_value_overrides_to_non_canonical_profiles.rb

```diff
@@ -0,0 +1,16 @@
+class ValueOverridesToNonCanonicalProfiles < ActiveRecord::Migration[7.0]
+  class Profile < ApplicationRecord; end
+
+  def up
+    profiles = Arel::Table.new(:profiles)
+    parent_profiles = Arel::Table.new(:profiles, :as => "parent_profiles")
+
+    update_manager = Arel::UpdateManager.new(ActiveRecord::Base)
+    update_manager.table(profiles)
+    update_manager.set([[profiles[:value_overrides], parent_profiles[:value_overrides]]])
+
+    sub_query = profiles[:parent_profile_id].not_eq(nil).and(profiles[:parent_profile_id].eq(parent_profiles[:id]))
+    query = "#{update_manager.to_sql} FROM profiles AS parent_profiles WHERE #{sub_query.to_sql}"
+    ActiveRecord::Base.connection.execute(query)
```

Comment:
This is the PostgreSQL update/join syntax:
```
UPDATE t1
SET t1.c1 = new_value
FROM t2
WHERE t1.c2 = t2.c2;
```

and this is what this is the query generated by what I have here:
```
UPDATE "profiles" 
SET "value_overrides" = "parent_profiles"."value_overrides"
FROM profiles AS parent_profiles
WHERE "profiles"."parent_profile_id" IS NOT NULL AND "profiles"."parent_profile_id" = "parent_profiles"."id"
```